### ISSUE TITLE
RaggedTensor support for model output

### ISSUE THREAD

**Issue by tungnat97** [tensorflow/tensorflow#43093]
**System information**
- TensorFlow version: 2.3.0
- Are you willing to contribute it (Yes/No): No

**Current behavior/state.**
Loss function is implemented in a way that tries to convert `y_true` and `y_pred` to Tensor. That means the output of a model can't be a RaggedTensor, otherwise error shows up as RaggedTensor can't be converted to Tensor.

**Will this change the current api? How?**
Line 62 of `/tensorflow/python/ops/confusion_matrix.py`: `predictions = ops.convert_to_tensor(predictions)`.
Line 63 of `/tensorflow/python/ops/confusion_matrix.py`: `labels = ops.convert_to_tensor(labels)`.
There should be an option to work with `labels` and `predictions` as RaggedTensor instead of just converting it to Tensor.

**Who will benefit with this feature?**
Sequence-to-sequence autoencoders with RaggedTensor inputs and outputs or models with RaggedTensor outputs in general will benefit.

**Comment by foxik** [comment]
Note that supporting RaggedTensors as targets (e.g., in `model.fit`) is a frequent request, see #44988, #44112, #43591, #43093, #42320, #41810.

Some partial progress is in pull requests #45060 and #45015, which will allow using custom losses in Keras model (but existing losses like MSE or (S)CE will still not work).

**Comment by foxik** [comment]
Note that I created a new feature request #45403 to support RaggedTensors in standard Keras loss functions.

**Comment by rmothukuru** [comment]
@tungnat97,
As the issue, #45403 has been resolved, can you please confirm if we can close this issue? Thanks!
